test/tic_tac_toe/setup_test.clj

```diff
@@ -1,21 +1,45 @@
 (ns tic-tac-toe.setup-test
   (:use [clojure.test :refer :all]
         [clojure.java.io :only [delete-file]]
+        [tic-tac-toe.mocks.mock-io :as io]
+        [tic-tac-toe.mocks.mock-ui :as ui]
         [tic-tac-toe.mocks.mock-serializer :as mock-serializer]
         [tic-tac-toe.player :as player]
         [tic-tac-toe.read-write.reader :as reader]
         [tic-tac-toe.setup :refer :all]))
 
-(def board ["X", 'empty, 'empty, 'empty,"O", 'empty, 'empty, 'empty, 'empty])
+(def board (vec (repeat 9 'empty)))
+(def board-moves ["X", 'empty, 'empty, 'empty,"O", 'empty, 'empty, 'empty, 'empty])
+
+(def new-game-io (io/mock-value-output "1" ""))
+(def new-game-ui (ui/create-mock-ui new-game-io))
+(def saved-game-io (io/mock-value-output "2" ""))
+(def saved-game-ui (ui/create-mock-ui saved-game-io))
 (def test-serializer (mock-serializer/create-mock-serializer))
 (def test-reader (create-reader test-serializer))
 
-(deftest sets-up-game-pieces
-  (testing "returns hash with board and players"
-    (let [game (setup-game "test/saves" "test")]
-      (is (= tic_tac_toe.user_interface.ConsoleUI (class (:ui game))))
-      (is (= tic_tac_toe.read_write.reader.FSReader (class (:reader game))))
-      (is (= tic_tac_toe.read_write.writer.FSWriter (class (:writer game))))
+(def new-game-utils (fn [] (hash-map :ui new-game-ui :reader test-reader)))
+(def saved-game-utils (fn [] (hash-map :ui saved-game-ui :reader test-reader)))
+
+(deftest sets-up-game-utils
+  (testing "returns hash with correct game utils"
+    (let [utils (setup-utils)]
+      (is (= tic_tac_toe.user_interface.ConsoleUI (class (:ui utils))))
+      (is (= tic_tac_toe.read_write.reader.FSReader (class (:reader utils))))
+      (is (= tic_tac_toe.read_write.writer.FSWriter (class (:writer utils)))))))
```

Comment:
Is there another way you can test this without checking the type?

Reply:
[Adjusted this](https://github.com/saralein/tic-tac-toe_clojure/pull/10/files#diff-15f263ec5b23d4d6b199b76016540567R1).